### PR TITLE
feat: adds ability to exclude vars by name

### PR DESCRIPTION
**Pull request by jurajpiar**
Adds a simple array of var names to exclude from the var lookup.

TODO: extend filter to include contract names for better filtering.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -3,3 +3,4 @@ node_modules
 /artifacts
 /cache
 /contracts-exposed
+.DS_Store
```
Comment (frangio) [inline .gitignore:6]: You should add this in a system-wide gitignore file rather than in each project.

**Comment by frangio** [comment]
Thanks @jurajpiar.

You seem to have run a code formatter on the whole codebase. Can you undo those changes?

Partly because of the above, I don't understand what this PR is adding. Can you explain further?

File: .gitignore (hunk 1 above)
Comment (jurajpiar) [reply .gitignore:6]: oops. will do yea tx

**Comment by jurajpiar** [comment]
> Thanks @jurajpiar.
> 
> You seem to have run a code formatter on the whole codebase. Can you undo those changes?
> 
> Partly because of the above, I don't understand what this PR is adding. Can you explain further?

Hi, sorry about the formatting. I'll revert that.
The idea of this PR is to allow exclusion of some internal variables from exposing. The motivation for it is to prevent compilation errors for generated exposed getters for storage variable, such as mappings inside structs.

As mentioned in the description, this implementation could be improved to allow definition of contracts in which these named methods would be excluded.

Hope that makes sense?

**Comment by frangio** [comment]
If this is about compilation errors, I'd favor an approach where we can detect the scenarios that would lead to compilation errors and ignores those variables.

**Comment by jurajpiar** [comment]
yea makes sense

I only needed to make this work for our project and with a tight deadline I did what found the easiest

once I have little bit more time (should be soon) I'll try to detect these instead.

**Pull request closed without merging**